server/src-lib/Hasura/Eventing/EventTrigger.hs

```diff
@@ -215,28 +141,52 @@ processEvent logenv pool getSchemaCache e = do
       let webhook = T.unpack $ wciCachedValue $ etiWebhookInfo eti
           retryConf = etiRetryConf eti
           timeoutSeconds = fromMaybe defaultTimeoutSeconds (rcTimeoutSec retryConf)
-          responseTimeout = HTTP.responseTimeoutMicro (timeoutSeconds * 1000000)
+          respTimeout = HTTP.responseTimeoutMicro (timeoutSeconds * 1000000)
           headerInfos = etiHeaders eti
           etHeaders = map encodeHeader headerInfos
           headers = addDefaultHeaders etHeaders
           ep = createEventPayload retryConf e
-      res <- runExceptT $ tryWebhook headers responseTimeout ep webhook
+          extraLogCtx = ExtraLogContext (epId ep)
+      res <- runExceptT $ withEventEngineCtx eventEngineCtx $
+        tryWebhook headers respTimeout (toJSON ep) webhook (Just extraLogCtx)
       let decodedHeaders = map (decodeHeader logenv headerInfos) headers
       finally <- either
         (processError pool e retryConf decodedHeaders ep)
         (processSuccess pool e decodedHeaders ep) res
       either logQErr return finally
 
+withEventEngineCtx ::
+  ( MonadReader r m
+  , Has HTTP.Manager r
+  , Has (L.Logger L.Hasura) r
+  , MonadIO m
+  , MonadError HTTPErr m
+  )
+  => EventEngineCtx
+  -> m HTTPResp
+  -> m HTTPResp
+withEventEngineCtx eeCtx = bracket_ (incrementThreadCount eeCtx) (decrementThreadCount eeCtx)
```

Comment:
@lexi-lambda  I was trying to use the bracket pattern here but it seems it only works with IO a types. Any recommendations?

Reply:
[`MonadMask`](https://hackage.haskell.org/package/exceptions-0.10.4/docs/Control-Monad-Catch.html#t:MonadMask) generalizes `bracket` to other monads.

Reply:
Well, that was easy !